python/ray/dashboard/subprocesses/module.py

```diff
@@ -84,78 +78,95 @@ async def init(self):
+            lambda x: (
+                inspect.ismethod(x)
+                and hasattr(x, "__route_method__")
+                and hasattr(x, "__route_path__")
+            ),
+        )
+        for _, handler in handlers:
+            routes.append(
+                aiohttp.web.route(
+                    handler.__route_method__,
+                    handler.__route_path__,
+                    handler,
                 )
+            )
+        app.add_routes(routes)
+        runner = aiohttp.web.AppRunner(app)
+        await runner.setup()
 
-    async def _internal_health_check(
-        self, message: RequestMessage, parent_bound_queue: multiprocessing.Queue
-    ) -> None:
-        """
-        Internal health check. Sends back a response to the parent queue.
+        socket_path = os.path.join(self._config.socket_dir, self.__class__.__name__)
+        if sys.platform == "win32":
+            site = aiohttp.web.NamedPipeSite(runner, socket_path)
+        else:
+            site = aiohttp.web.UnixSite(runner, socket_path)
+        await site.start()
+        logger.info("Started aiohttp server.")
 
-        Note this is NOT registered as a route, so an external HTTP request will not
-        trigger this.
+    @abc.abstractmethod
+    async def is_healthy(self) -> bool:
+        """
+        Whether the module is considered healthy.
+        Subclasses should implement this method to provide health check logic.
         """
+        pass
+
+    async def _internal_module_health_check(self, request):
         try:
-            parent_bound_queue.put(
-                UnaryResponseMessage(
-                    request_id=message.request_id, status=200, body=b"ok!"
-                )
-            )
-        except Exception as e:
-            logger.error(
-                f"Error sending response: {e}. This means we will never reply the parent's health check request. The parent will think the module is dead."
+            is_healthy = await self.is_healthy()
+        except Exception:
+            logger.exception("Error during internal health check")
+            is_healthy = False
+        if is_healthy:
+            return aiohttp.web.Response(
+                text="success",
+                content_type="application/text",
             )
+        return aiohttp.web.HTTPServiceUnavailable(reason="Internal health check failed")
+
+
+async def run_module_inner(
+    cls: type[SubprocessModule],
+    config: SubprocessModuleConfig,
+    ready_event: multiprocessing.Event,
+):
+
+    module_name = cls.__name__
+
+    logger.info(f"Starting module {module_name} with config {config}")
+
+    try:
+        module = cls(config)
+        # First init the module, then start the aiohttp server.
+        await module.init()
+        await module.start_server()
+        ready_event.set()
+        logger.info(f"Module {module_name} initialized, receiving messages...")
+        while True:
+            await asyncio.sleep(3600)  # sleep forever
```

Comment:
Do we need this given we call `loop.run_forever()` later

Reply:
Removed